tests/common/fixtures/consistency_checker/consistency_checker.py

```diff
@@ -0,0 +1,346 @@
+import pytest
+import logging
+import json
+import os
+import datetime
+from collections import defaultdict
+from tests.common.fixtures.consistency_checker.constants import SUPPORTED_PLATFORMS_AND_VERSIONS
+
+logger = logging.getLogger(__name__)
+
+SYNCD_CONTAINER = "syncd"
+QUERY_ASIC_SCRIPT = "query-asic.py"
+QUERY_ASIC_PARSER = "parser.py"
+LIBSAIREDIS_DEB = "libsairedis.deb"
+PYTHON3_PYSAIREDIS_DEB = "python3-pysairedis.deb"
+DUT_DST_PATH_HOST = "/tmp/consistency-checker"
+DUT_DST_PATH_CONTAINER = "/consistency-checker"
+
+QUERY_ASIC_PATH_SRC = os.path.dirname(__file__) + "/query-asic"
+QUERY_ASIC_SCRIPT_PATH_SRC = QUERY_ASIC_PATH_SRC + "/" + QUERY_ASIC_SCRIPT
+QUERY_ASIC_PARSER_PATH_SRC = QUERY_ASIC_PATH_SRC + "/" + QUERY_ASIC_PARSER
+QUERY_ASIC_SCRIPT_PATH_DST_HOST = DUT_DST_PATH_HOST + "/" + QUERY_ASIC_SCRIPT
+QUERY_ASIC_PARSER_PATH_DST_HOST = DUT_DST_PATH_HOST + "/" + QUERY_ASIC_PARSER
+QUERY_ASIC_SCRIPT_PATH_DST_CONTAINER = DUT_DST_PATH_CONTAINER + "/" + QUERY_ASIC_SCRIPT
+
+LIBSAIREDIS_TEMP = "libsairedis-temp"
+
+
+class ConsistencyChecker:
+
+    def __init__(self, duthost, libsairedis_download_url=None, python3_pysairedis_download_url=None):
+        """
+        If the libsairedis_download_url and python3_pysairedis_download_url are provided, then these artifacts
+        are downloaded and installed on the DUT, otherwise it's assumed that the environment is already setup
+        for the consistency checker.
+        """
+        self._duthost = duthost
+        self._libsairedis_download_url = libsairedis_download_url
+        self._python3_pysairedis_download_url = python3_pysairedis_download_url
+
+    def __enter__(self):
+        logger.info("Initializing consistency checker on dut...")
+
+        self._duthost.shell(f"mkdir -p {DUT_DST_PATH_HOST}")
+        self._duthost.copy(src=QUERY_ASIC_SCRIPT_PATH_SRC, dest=QUERY_ASIC_SCRIPT_PATH_DST_HOST)
+        self._duthost.copy(src=QUERY_ASIC_PARSER_PATH_SRC, dest=QUERY_ASIC_PARSER_PATH_DST_HOST)
+
+        if self._libsairedis_download_url is not None:
+            self._duthost.shell(f"curl -o {DUT_DST_PATH_HOST}/{LIBSAIREDIS_DEB} {self._libsairedis_download_url}")
+        if self._python3_pysairedis_download_url is not None:
+            self._duthost.shell(
+                f"curl -o {DUT_DST_PATH_HOST}/{PYTHON3_PYSAIREDIS_DEB} {self._python3_pysairedis_download_url}")
+
+        # Move everything into syncd container
+        self._duthost.shell((
+            f"docker cp {DUT_DST_PATH_HOST} {SYNCD_CONTAINER}:/ && "
```

Comment:
Non-blocking comment: I'll need to recheck later, but we might not actually need to touch the syncd container at all, and instead might be able to run everything from the host itself. This may require extracting/installing another package or two, though.

Reply:
I don't recall which packages but there were a few more that needed to be installed on the host but were already present in the container.